src/sas/qtgui/Calculators/Shape2SAS/DesignWindow.py

```diff
@@ -349,7 +360,12 @@ def ifFitPar(value: Union[float, str], values: list[Union[float, str]], *args, *
         row, column = args #int, int
         
         if not value == "":
-            if conditionBool[column][row]:
+            # Add bounds checking to prevent IndexError
+            if (column < len(conditionBool) and 
+                row < len(conditionBool[column]) and 
+                column < len(conditionFitPar) and 
+                row < len(conditionFitPar[column]) and
+                conditionBool[column][row]):
```

Comment:
Is there a valid reason column and row would be out of bounds?

If there is a valid reason fine (what is it?). If not this should either be removed and allow a traceback that can lead us to the actual error or raise an exception. Masking errors seems like very bad programming form that will end up biting us hard later